### PR TITLE
refactor: move all scripts into package.json

### PR DESCRIPTION
**Pull request by roedoejet**
also moves dependencies into the main package.json so that we don't need to cd anywhere.

I think I prefer keeping everything in npm (even if it's just calling a bash script) since it unifies the commands that users have to type. I also prefer command line arguments over environment variables as a general rule. 

I also prefer linking the dependencies like this instead of instructing the user to cd into the project.

Instead of allowing an arbitrary data path, I think I also prefer just parameterizing around the language id and version.

### CODE REVIEW

File: projects/word-weaver-cli/validate.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import * as fs from "fs";
 import Ajv from "ajv";
-import { env } from "process";
+import { argv } from "process";
 
-const DATA_DIR = env.DATA_DIR || "../word-weaver/src/assets/data/fr/v1";
+const DATA_LANG = argv[2] || "fr";
```
Comment (joanise) [inline projects/word-weaver-cli/validate.ts:5]: I'm surprised this is argv 2 and 3, not 0 and 1. You tested this?

File: projects/word-weaver-cli/gen-schemas.sh [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
-#    DATA_DIR=../word-weaver/src/assets/data/fr/v1 ./compress.sh
+#    ./gen-schemas.sh fr
 
 set -o errexit
 
 CONFIG_PATH=../word-weaver/src/config
-if [[ $SCHEMA_LANG ]]; then
-    CONFIG_FILE=schema."$SCHEMA_LANG".ts
+if [[ $1 ]]; then
+    CONFIG_FILE=schema."$1".ts
```
Comment (joanise) [inline projects/word-weaver-cli/gen-schemas.sh]:
```suggestion
SCHEMA_LANG=$1
if [[ $SCHEMA_LANG ]]; then
    CONFIG_FILE=schema."$SCHEMA_LANG".ts
```
just to make it more self-documenting

**Review by joanise — APPROVED** [review]
This makes sense, it will be more intuitive, and I like you link it up to the top.
I assume you've tested all this and it works, with and without the optional parameters?

I'm clicking approve, but there is a small change request.

File: projects/word-weaver-cli/validate.ts (hunk 1 above)
Comment (roedoejet) [reply projects/word-weaver-cli/validate.ts:5]: yep - 0 is the entry point and 1 is the script in node's argv

File: projects/word-weaver-cli/gen-schemas.sh (hunk 2 above)
Comment (roedoejet) [reply projects/word-weaver-cli/gen-schemas.sh]: yep - this is better, go for it!

**Review by joanise — APPROVED** [review]

**Pull request merged**